Frends.HTTP.DownloadFile/Frends.HTTP.DownloadFile/DownloadFile.cs

```diff
@@ -0,0 +1,143 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+[assembly: InternalsVisibleTo("Frends.HTTP.DownloadFile.Tests")]
+namespace Frends.HTTP.DownloadFile;
+
+/// <summary>
+/// HTTP Task.
+/// </summary>
+public static class HTTP
+{
+    internal static IHttpClientFactory ClientFactory = new HttpClientFactory();
+    internal static readonly ObjectCache ClientCache = MemoryCache.Default;
+    private static readonly CacheItemPolicy _cachePolicy = new() { SlidingExpiration = TimeSpan.FromHours(1) };
+
+    internal static void ClearClientCache()
+    {
+        var cacheKeys = ClientCache.Select(kvp => kvp.Key).ToList();
+        foreach (var cacheKey in cacheKeys)
+            ClientCache.Remove(cacheKey);
+    }
+
+    /// For mem cleanup.
+    static HTTP()
+    {
+        var currentAssembly = Assembly.GetExecutingAssembly();
+        var currentContext = AssemblyLoadContext.GetLoadContext(currentAssembly);
+        if (currentContext != null)
+            currentContext.Unloading += OnPluginUnloadingRequested;
+    }
+
+    /// <summary>
+    /// Fetch file over HTTP.
+    /// [Documentation](https://tasks.frends.com/tasks/frends-tasks/Frends.HTTP.DownloadFile)
+    /// </summary>
+    /// <param name="input">Input parameters</param>
+    /// <param name="options">Optional parameters.</param>
+    /// <param name="cancellationToken">Token generated by frends to stop this Task.</param>
+    /// <returns>Object { bool Success, string FilePath } }</returns>
+    public static async Task<Result> DownloadFile([PropertyTab] Input input, [PropertyTab] Options options, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrEmpty(input.Url)) throw new ArgumentNullException(input.Url);
+
+        try
+        {
+            var httpClient = GetHttpClientForOptions(options);
+            var headers = GetHeaderDictionary(input.Headers, options);
+
+            //Clear default headers
+            httpClient.DefaultRequestHeaders.Clear();
+
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                {
+                    var requestHeaderAddedSuccessfully = httpClient.DefaultRequestHeaders.TryAddWithoutValidation(header.Key, header.Value);
+                    if (!requestHeaderAddedSuccessfully)
+                        Trace.TraceWarning($"Could not add header {header.Key}:{header.Value}");
+                }
+            }
+
+            using var s = await httpClient.GetStreamAsync(input.Url, cancellationToken);
+            using var fs = new FileStream(input.FilePath, FileMode.CreateNew);
+            await s.CopyToAsync(fs, cancellationToken);
+
+            return new Result(true, fs.Name);
+        }
+        catch (Exception ex)
+        {
+            throw new Exception(ex.Message, ex.InnerException);
+        }
+    }
+
+    private static IDictionary<string, string> GetHeaderDictionary(Header[] headers, Options options)
+    {
+        if (headers != null)
```

Comment:
Not sure if this works, since Frends never passes the Header array as null. If parameters are not set, it will pass an empty Header array. So you should change this check to `headers.Length > 0`.